syncd/syncd.cpp

```diff
@@ -554,26 +859,34 @@ int main(int argc, char **argv)
     // also "remove" from response queue will also trigger another "response"
     getRequest = new swss::ConsumerTable(db, "GETREQUEST");
     getResponse  = new swss::ProducerTable(db, "GETRESPONSE");
+    notifications = new swss::ProducerTable(dbNtf, "NOTIFICATIONS");
 
     sai_api_initialize(0, (service_method_table_t*)&test_services);
 
     populate_sai_apis();
 
     initialize_common_api_pointers();
 
-    // swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_DEBUG);
-
 #if 1
+
     sai_status_t status = sai_switch_api->initialize_switch(0, "0xb850", "", &switch_notifications);
 
     if (status != SAI_STATUS_SUCCESS)
     {
-        SYNCD_LOG_ERR("fail to sai_initialize_switch: %lld", status);
+        SWSS_LOG_ERROR("fail to sai_initialize_switch: %d", status);
         exit(EXIT_FAILURE);
     }
 
-    std::thread bcm_diag_shell_thread = std::thread(sai_diag_shell);
-    bcm_diag_shell_thread.detach();
+    for (int i = 0; i < argc; i++)
+    {
+        if (strcmp(argv[i],"--diag") == 0)
```

Comment:
Yes, i wanted to join hrachyam since he should work on that so i wanted to switch to getopt when his implementation would be available